vizro-core/src/vizro/models/_controls/filter.py

```diff
@@ -124,6 +147,31 @@ def _set_slider_values(self):
             if self.selector.max is None:
                 self.selector.max = max(max_values)
 
+    def _set_date_picker_values(self):
+        if isinstance(self.selector, SELECTORS["temporal"]):
+            if self._column_type != "temporal":
+                raise ValueError(
+                    f"Chosen selector {self.selector.type} is not compatible "
+                    f"with {self._column_type} column '{self.column}'."
+                )
+
+            min_values = []
+            max_values = []
+            for target_id in self.targets:
+                data_frame = data_manager._get_component_data(target_id)
+                data_frame = self._convert_column_type(data_frame=data_frame)
```

Comment:
This can probably be done in a nicer way once the data refreshes are possible in the data manager. Would we store the converted data or do you think we shouldn't do that anyways @antonymilne ?

Reply:
Hmm good question, I'm not sure. Definitely it feels a bit wrong having two calls to `_convert_column_type` at the moment, but also I'm not sure we should be persisting changes to data. So let's leave it this way for now.